commit fdf1e04094fe525ca6361ecbd459f69932d6f2e0
Author: Fabio Erculiani <lxnay@sabayon.org>
Date:   Wed Sep 1 17:15:57 2010 +0200

    entropy: encode input strings into UTF-8 before processing, thanks to Daniel Halens for reporting

diff --git a/backends/entropy/entropyBackend.py b/backends/entropy/entropyBackend.py
index f410b73..d3a4bf4 100755
--- a/backends/entropy/entropyBackend.py
+++ b/backends/entropy/entropyBackend.py
@@ -109,6 +109,10 @@ class PackageKitEntropyMixin(object):
                 ' '.join([const_convert_to_unicode(x) for x in my_args]),)
             )
 
+    def _encode_string_list(self, values):
+        # values is a list of unencoded strings, we need UTF-8 strings here
+        return [const_convert_to_unicode(x) for x in values]
+
     def _is_repository_enabled(self, repo_name):
         """
         Return whether given repository identifier is available and enabled.
@@ -1453,6 +1457,8 @@ class PackageKitEntropyBackend(PackageKitBaseBackend, PackageKitEntropyMixin):
 
     def search_details(self, filters, values):
 
+        values = self._encode_string_list(values)
+
         self._log_message(__name__, "search_details: got %s and %s" % (
             filters, values,))
 
@@ -1490,6 +1496,8 @@ class PackageKitEntropyBackend(PackageKitBaseBackend, PackageKitEntropyMixin):
 
     def search_file(self, filters, values):
 
+        values = self._encode_string_list(values)
+
         self._log_message(__name__, "search_file: got %s and %s" % (
             filters, values,))
 
@@ -1548,6 +1556,8 @@ class PackageKitEntropyBackend(PackageKitBaseBackend, PackageKitEntropyMixin):
 
     def search_group(self, filters, values):
 
+        values = self._encode_string_list(values)
+
         self._log_message(__name__, "search_group: got %s and %s" % (
             filters, values,))
 
@@ -1611,6 +1621,8 @@ class PackageKitEntropyBackend(PackageKitBaseBackend, PackageKitEntropyMixin):
 
     def search_name(self, filters, values):
 
+        values = self._encode_string_list(values)
+
         self._log_message(__name__, "search_name: got %s and %s" % (
             filters, values,))
 
